punishments.js

```diff
@@ -843,6 +845,56 @@ Punishments.unnamelock = function (name) {
 	}
 	return success;
 };
+/**
+ * @param {User} user
+ * @param {number} expireTime
+ * @param {string} id
+ * @param {...string} [reason]
+ * @return {PunishmentRow}
+ */
+Punishments.battleban = function (user, expireTime, id, ...reason) {
```

Comment:
I'm pretty sure a user can only have 1 global punishment at a time.

I feel that having a locked user also being battle banned should be allowed, though it might be tricky in this case. We may have to just global ban in those situations.

Reply:
oh I didn't know that. Yeah I wanted it so battle ban and lock were separate so users could still chat in chatrooms if they got battlebanned, but I suppose you should just be locked and battlebanned instead if you're bad enough to warrant one.

Reply:
Although having it done the intended way would be preferred, how would I go about doing that ?

Reply:
Battle bans should act like namelocks where they also cannot talk, that could solve this issue

Reply:
That’s what we said, is that how everyone feels? Hoeen and I both agree there’s a reason to be banned from battling and not locked at the same time. But honestly I’d be fine with either option